functions/Test-DbaDbQueryStore.ps1

```diff
@@ -0,0 +1,242 @@
+            }
+
+            if ($Database) {
+                $dbDatabases = $dbDatabases | Where-Object { $Database -contains $_.Name }
+            }
+
+            if ($ExcludeDatabase) {
+                $dbDatabases = $dbDatabases | Where-Object Name -NotIn $ExcludeDatabase
+            }
+
+            $desiredState = [PSCustomObject]@{
+                Property      = 'ActualState'
+                Value         = 'ReadWrite'
+                Justification = 'This means Query Store is enabled and collecting data.'
+            },
+            [PSCustomObject]@{
+                Property      = 'DataFlushIntervalInSeconds'
+                Value         = '900'
+                Justification = 'Recommended to leave this at the default of 900 seconds (15 mins).'
+            },
+            [PSCustomObject]@{
+                Property      = 'MaxPlansPerQuery'
+                Value         = '200'
+                Justification = 'Number of distinct plans per query. 200 is a good starting point for most environments.'
+            },
+            [PSCustomObject]@{
+                Property      = 'MaxStorageSizeInMB'
+                Value         = '2048'
+                Justification = 'How much disk space Query Store will use. 2GB is a good starting point.'
+            },
+            [PSCustomObject]@{
+                Property      = 'QueryCaptureMode'
+                Value         = 'Auto'
+                Justification = 'With auto, queries that are insignificant from a resource utilization perspective, or executed infrequently, are not captured.'
+            },
+            [PSCustomObject]@{
+                Property      = 'SizeBasedCleanupMode'
+                Value         = 'Auto'
+                Justification = 'With auto, as Query Store gets close to out of space it will automatically purge older data.'
+            },
+            [PSCustomObject]@{
+                Property      = 'StaleQueryThresholdInDays'
+                Value         = '30'
+                Justification = 'Determines how much historic data to keep. 30 days is a good value here.'
+            },
+            [PSCustomObject]@{
+                Property      = 'StatisticsCollectionIntervalInMinutes'
+                Value         = '30'
+                Justification = 'Time window that runtime stats will be aggregated. Use 30 unless you have space concerns, then leave at the default (60).'
+            },
+            [PSCustomObject]@{
+                Property      = 'WaitStatsCaptureMode'
+                Value         = 'ON'
+                Justification = 'Adds valuable data when troubleshooting.'
+            }
+
+            try {
+                Write-Message -Level Verbose -Message "Evaluating Query Store options"
+                $currentOptions = Get-DbaDbQueryStoreOption -SqlInstance $smo -Database $dbDatabases.name
+
+                $currentOptions |
```

Comment:
The formatting issue is coming from this in most situations (happened on another PR as well).

I would look at rewriting this without needing so much piping use.